# This is a GitHub Repository of code used for the publication titled: Spatial and finely tuned temporal metagenomics of river compartments reveals viral community dynamics in an urban stream

The different files in this repository have all the necessary steps that are required to re-create the output that we used to write this manuscript. In each file, you will either find specific commands on what exactly was run, and how to get input files for them if they are part of a pipeline, OR, they will contain commands as well as the input files that were used (e.g., 2.1..., 2.2...3.1...) where each whole number corresponds to the main command and the decimals represent the additional files provided.


## Manuscript abstract:

Although river ecosystems comprise less than 1% of Earth’s total non-glaciated area, they are critical modulators of microbially and virally orchestrated global biogeochemical cycles. However, in the field of river microbial ecology, most studies either use data that is not spatially resolved or is collected at timepoints that do not reflect the short life cycles of these microorganisms. As a result, the relevance of microbiome interactions and the impacts they have over time on biogeochemical cycles are poorly constrained in models. To assess how viral communities and their hosts change over time, we sampled surface-water and pore-water from the Erpe river, Germany every 3 hours over a 48-hour period resulting in 32 metagenomes paired to geochemical and metabolite measurements. We reconstructed 6500 viral and 1033 microbial genomes and found distinct communities associated with each river compartment that reflected metabolisms consistent with inferred redox conditions (e.g., methanogenesis in pore-water, photosynthesis in surface-water). Our results show that 70% of the viral community was persistent in surface-water samples, whereas only 13% were persistent in the pore-water samples. Finally, we predicted linkages between 73 viral genomes and 38 microbial genomes. These putatively linked hosts were members of the Competibacteraceae, which we show are key contributors to carbon and nitrogen cycling in rivers. Together, these findings bolster support for the idea that microbial and viral communities can exist as taxonomically, and metabolically stable communities along a flowing river corridor; and raise important questions that affect ecosystem models attempting to constrain the underpinning dynamics of biogeochemical cycles in rivers.

## Author list:

Josué Rodríguez-Ramos1, Angela Oliverio1,2, Mikayla A. Borton1,3, Birgit M. Mueller4, Hanna Schulz4, Rory M. Flynn1, Rebecca A. Daly1, Robert Danczak3, Jared Ellenbogen1, LeAundra Schopflin1, Michael Shaffer1, Amy Goldman3, Joerg Lewandowski4, James C. Stegen3, Kelly C. Wrighton1

1 Department of Soil and Crop Sciences, Colorado State University, Fort Collins, CO, USA
2 Department of Biology, Syracuse University, Syracuse NY, USA
3 Biological Sciences Division, Pacific Northwest National Laboratory, Richland, WA, USA
4 Leibniz Institute of Freshwater Ecology and Inland Fisheries, Berlin, Germany

\*Corresponding author: Kelly Wrighton

<img width="468" alt="image" src="https://user-images.githubusercontent.com/95941755/223815670-e531cf43-a566-4179-bdca-f8b44dd0ce4f.png">
Figure 1: Experimental design enables a genome- and time-resolved view of microbial communities at a finely scaled resolution. a) Satellite image of the Erpe river sampling site that is located near Berlin, Germany. b) Conceptual schematic of the surface and porewater sections that were collected as part of this research. c) Data types that were collected as part of this sampling effort. d) Schematic across time of two main ecological variables (water stage, and temperature) across the timepoints collected. The colors and icons highlight the stage of the day when samples were collected.

