# This is a GitHub Repository of code used for the publication titled: Finely tuned spatial and temporal metagenomics of river compartments reveals viral communities can be stable, cosmopolitan, and impact microbial community function across time

The different files in this repository have all the necessary steps that are required to re-create the output that we used to write this manuscript. In each file, you will either find specific commands on what exactly was run, and how to get input files for them if they are part of a pipeline, OR, they will contain commands as well as the input files that were used (e.g., 2.1..., 2.2...3.1...) where each whole number corresponds to the main command and the decimals represent the additional files provided.


## Manuscript abstract:

Although river ecosystems comprise less than 1% of Earth’s total non-glaciated area, they are critical modulators of microbially and virally orchestrated global biogeochemical cycles. However, in the field of river microbial ecology, most studies either use data that is not spatially resolved or is collected at timepoints that do not reflect the short life cycles of these microorganisms. As a result, the relevance of microbiome interactions and the impacts they have over time on biogeochemical cycles are poorly constrained in models. To assess how viral communities and their hosts change over time, we sampled surface-water and pore-water from the Erpe river, Germany every 3 hours over a 48-hour period resulting in 32 metagenomes paired to geochemical measurements. We reconstructed 6500 viral and 1033 microbial genomes and found distinct communities associated with each compartment that reflected their metabolisms (e.g., methanogenesis in pore-water, photosynthesis in surface-water). Our results show that 77% of the viral community was persistent in surface-water samples, whereas 70% were intermittent or ephemeral in pore-water samples. Finally, we predicted linkages between 128 viral genomes and 45 microbial genomes. Among these putatively linked hosts were members of the Actinobacteriota, Campylobacter, and Halobacteriota which are key contributors to carbon, nitrogen, and sulfur cycling in rivers. Together, these findings bolster support for the idea that microbial and viral communities can exist as taxonomically, and metabolically stable communities along a flowing river corridor; and raise important questions that could affect ecosystem models attempting to constrain the underpinning dynamics of biogeochemical cycles in rivers.

## Author list:

Josué Rodríguez-Ramos1, Angela Oliverio 1, Mikayla A. Borton1,2, Rebecca A. Daly1, Birgit M. Mueller3, LeAundra Schopflin1, Michael Shaffer1, Robert Danczak2, Hanna Schulz3, Simon Roux4, Amy Goldman2, Joerg Lewandowski3, James C. Stegen2, Kelly C. Wrighton1

1 Department of Soil and Crop Sciences, Colorado State University, Fort Collins, CO, USA
2 Biological Sciences Division, Pacific Northwest National Laboratory, Richland, WA, USA
3 Leibniz Institute of Freshwater Ecology and Inland Fisheries, Berlin, Germany
4 Joint Genome Institute, Viral Genomics Group, Berkeley California

\*Corresponding author: Kelly Wrighton
